rewrite-core/src/main/java/org/openrewrite/GitRemote.java

```diff
@@ -47,69 +48,116 @@ public enum Service {
     }
 
     public static class Parser {
-        private final Map<String, Service> origins;
+        private final List<RemoteServer> servers;
 
         public Parser() {
-            origins = new LinkedHashMap<>();
-            origins.put("github.com", Service.GitHub);
-            origins.put("gitlab.com", Service.GitLab);
-            origins.put("bitbucket.org", Service.BitbucketCloud);
-            origins.put("dev.azure.com", Service.AzureDevOps);
-            origins.put("ssh.dev.azure.com", Service.AzureDevOps);
+            servers = new ArrayList<>();
+            servers.add(new RemoteServer(Service.GitHub, "github.com", URI.create("https://github.com"), URI.create("ssh://git@github.com")));
+            servers.add(new RemoteServer(Service.GitLab, "gitlab.com", URI.create("https://gitlab.com"), URI.create("ssh://git@gitlab.com")));
+            servers.add(new RemoteServer(Service.BitbucketCloud, "bitbucket.org", URI.create("https://bitbucket.org"), URI.create("ssh://git@bitbucket.org")));
+            servers.add(new RemoteServer(Service.AzureDevOps, "dev.azure.com", URI.create("https://dev.azure.com"), URI.create("ssh://git@ssh.dev.azure.com")));
         }
 
-        public Parser registerRemote(Service service, String origin) {
-            if (origin.startsWith("https://") || origin.startsWith("http://") || origin.startsWith("ssh://")) {
-                origin = new Parser.HostAndPath(origin).concat();
+        public URI toUri(GitRemote remote, boolean ssh) {
```

Comment:
Would it make sense to change the `ssh` parameter to `protocol` to allow for `http` or `https`?

Reply:
I did not think it was very useful, but it's easy to add now, harder to change later. So lets!